Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -124,6 +124,16 @@ extern ASCIILiteral errorAsString(Error);
 #define MESSAGE_CHECK_BASE(assertion, connection) MESSAGE_CHECK_COMPLETION_BASE(assertion, connection, (void)0)
 #define MESSAGE_CHECK_BASE_COROUTINE(assertion, connection) MESSAGE_CHECK_COMPLETION_BASE_COROUTINE(assertion, connection, (void)0)
 
+#define CONDITIONAL_MESSAGE_CHECK_BASE(assertion, connection, completion) do { \
+    if (UNLIKELY(!(assertion))) { \
+        if (RefPtr processConnection = connection) { \
+            RELEASE_LOG_FAULT(IPC, __FILE__ " " CONNECTION_STRINGIFY_MACRO(__LINE__) ": Invalid message dispatched %" PUBLIC_LOG_STRING, WTF_PRETTY_FUNCTION); \
```

Comment:
Should we consider logging the fault even if we don't have a connection to mark invalid? My mental model is that a message is either valid or not. But maybe there are some invalid messages that are OK if the process has exited?

Reply:
I think we can always log, yeah. I was just preserving the existing behavior but I don't see much reason not to log here.